src/main/pages/che-7/extensions/assembly_telemetry.adoc

```diff
@@ -0,0 +1,260 @@
+= Telemetry
+
+Telemetry is the transparent and ethical collection of the usage data. By default, telemetry is not available in the {prod}, 
+but there is an abstract API that allows enabling telemetry via the plugin mechanism. This approach is used in the link:{site-baseurl}che-7/hosted-che/[Hosted Che] service where telemetry is enabled for every workspace.
+
+== Use cases
+
+Eclipse Che telemetry API allows tracking:
+
+* Duration of workspace usage.
+* User-driven actions like file editing, committing and pushing to the remote repositories
+* The list of plugins enabled in the workspace
+* Programming languages and link:{site-baseurl}che-7/making-a-workspace-portable-using-a-devfile/devfiles[devfiles] used in the workspaces
+
+== How it works?
+
+When a {prod-short} workspace starts, the che-theia container starts the telemetry plugin, which is responsible for sending telemetry events to a backend.  If the `$CHE_WORKSPACE_TELEMETRY_BACKEND_PORT` environment variable is set in the workspace pod, the telemetry plugin will send events to a backend listening at that port.
+
+If the {prod-short} workspace has a telemetry backend container running, and it is listening on `$CHE_WORKSPACE_TELEMETRY_BACKEND_PORT` it will take those events sent from the telemetry plugin, turn them into the backend-specific representation of events, and send them to the configured analytics backend (e.g. Segment, Woopra, etc.).
+
+image::telemetry/telemetry_diagram.png[]
+
+We will create an `AnalyticsManager` class that extends link:https://github.com/che-incubator/che-workspace-telemetry-client/blob/master/backend-base/src/main/java/org/eclipse/che/incubator/workspace/telemetry/base/AbstractAnalyticsManager.java[`AbstractAnalyticsManager`] and implements the following methods:
+
+* `isEnabled()` - used to determine whether or not the telemetry backend is functioning correctly.  This could be as simple as always returning `true`, or have more complex checks, for example returning `false` when a connection property is missing.
+* `destroy()` - cleanup method that is run before shutting down the telemetry backend.  This method should send the `WORKSPACE_STOPPED` event.
+* `onActivity()` - notifies that some activity is still occuring for a given user. This is mainly used to send `WORKSPACE_INACTIVE` events.
+* `onEvent()` - submit telemetry events to the telemetry server, such as `WORKSPACE_USED`, `WORKSPACE_STARTED`, and so on.
+
+== Getting Started
+
+TODO: Describe the step-by-step flow for enabling telemetry via sidecar plugin for sending data to an external Analytics Service / Customer Data Platform
+
+This document will describe the steps required to extend the {prod-short} telemetry system to connect to a custom back-end:
+
+. Creating a server process that will receive events
+. Extending the {prod-short} libraries to create a backend that send events to the server
+. Adding a plugin for your backend and telling {prod-short} to load the plugin in your workspaces
```

Comment:
Might be a good idea to also clarify that backend should be delivered as an image that would be used in the plugin specification

Reply:
Fixed